packages/expo-image-picker/src/ExponentImagePicker.web.ts

```diff
@@ -141,3 +119,50 @@ function openFileBrowserAsync({
     input.dispatchEvent(event);
   });
 }
+
+
+
+async function readFile(targetFile: Blob): Promise<ImagePickerResult> {
+
+  return new Promise((resolve, reject) => {
+    const reader = new FileReader();
+    reader.onerror = () => {
+      reject(new Error(`Failed to read the selected media because the operation failed.`));
+    };
+    reader.onload = ({ target }) => {
+      const uri = (target as any).result;
+
+      const returnRaw = () => {
+        resolve({
+          cancelled: false,
+          uri,
+          width: 0,
+          height: 0,
+        });
+      };
+      if (typeof target?.result === 'string') {
+        const image = new Image();
+        image.src = target.result;
+        image.onload = function () {
+          resolve({
+            cancelled: false,
+            uri,
+            width: image.naturalWidth ?? image.width,
+            height: image.naturalHeight ?? image.height,
+          });
+        };
+        image.onerror = () => {
+          returnRaw();
+        };
+      } else {
+        returnRaw();
+      }
+    };
+    reader.readAsDataURL(targetFile);
+
+  })
+
+}
+
+
+
```

Comment:
Some styling suggestions:
```suggestion

async function readFile(targetFile: Blob): Promise<ImagePickerResult> {
  return new Promise((resolve, reject) => {
    const reader = new FileReader();
    reader.onerror = () => {
      reject(new Error(`Failed to read the selected media because the operation failed.`));
    };
    reader.onload = ({ target }) => {
      const uri = (target as any).result;
      const returnRaw = () => resolve({
        cancelled: false,
        uri,
        width: 0,
        height: 0,
      });
    
      if (typeof target?.result === 'string') {
        const image = new Image();
        image.src = target.result;
        image.onload = () => resolve({
          cancelled: false,
          uri,
          width: image.naturalWidth ?? image.width,
          height: image.naturalHeight ?? image.height,
        });
        image.onerror = () => returnRaw();
      } else {
        returnRaw();
      }
    };

    reader.readAsDataURL(targetFile);
  })
}
```